File: Company_wise_Mock/Microsoft/Delete Node in a Linkedlist.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node) {
        // the node to be deleted is given
        // for each node assign the next nodes val, delete the tail
        
        ListNode *parent = NULL;
        while(node){
            if(node->next)
            {
                node->val = node->next->val;
                parent = node;
                node = node->next;
            }
            
            else {
                // delete node
                if(parent) parent->next = NULL;
                delete node;
                break;
            }
        }
    }
};

File: June, 2020 Challenge/Delete Node in a Linked List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public void DeleteNode(ListNode node) {
        ListNode parent = node;
        while(node != null) {
            if(node.next == null) {
                parent.next = null;
                break;
            }
            
            else {
                node.val = node.next.val;
                parent = node;
                node = node.next;
            }
        }
    }
}
